File: app/src/main/java/co/medellinandroid/spotifyartists/di/module/AppModule.java
package co.medellinandroid.spotifyartists.di.module;

import co.medellinandroid.spotifyartists.MyApplication;
import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;

@Module
public class AppModule {

  MyApplication application;

  public AppModule(MyApplication application) {
    this.application = application;
  }

  @Provides @Singleton MyApplication providesApplication(){
    return application;
  }
}

File: app/src/main/java/co/medellinandroid/spotifyartists/di/module/AppModule.kt
package co.medellinandroid.spotifyartists.di.module

import co.medellinandroid.spotifyartists.MyApplication
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule(internal var application: MyApplication) {

  @Provides
  @Singleton internal fun providesApplication(): MyApplication = application
}
